### PR TITLE
Optimize the documentation JS files

### PR DESCRIPTION
**Pull request by RedYetiDev**
This PR optimizes the three files in the `/docs/_assets/js` directory for performance.

In `toolbar.js`, jQuery selectors have been cached as variables to prevent constant calls to the DOM.
In `sidebar.js`, the multiple `toggleClass(...).toggleClass(...)` have been merged into one
In `api-search.js`:
- Defined functions have been switched to newer ES format (`const myFunc = () => {}`)
- `regexForTerm` has been switched to ternary
- JSON has been switched to **JS**ON (`{"key":"value"}` -> `{key:"value"}`)

### CODE REVIEW

File: docs/_assets/js/api-search.js
Comment (jchyb) [inline docs/_assets/js/api-search.js:36]: I am not a fan of this syntax (I find it more difficult to read, the previous one is more similar to what we have in scala, and it adds some inconsistency to the code, with the other ifs being unchanged), and I doubt it brings any performance improvements. Could you use the previous one?

File: docs/_assets/js/toolbar.js
Comment (jchyb) [inline docs/_assets/js/toolbar.js:4]: The syntax changes to `function() {` here (and in other functions in this file) are a bit weird considering the opposite changes from `(args) => rhs` made in `api-search`. I personally prefer the `(args) => rhs` (since it's more similar to scala). Maybe let's keep the `function() {}` syntax for top level functions, and use lambda syntax for inner functions (like in `api-search`)?

Comment (RedYetiDev) [reply docs/_assets/js/toolbar.js:4]: Oh! Your right, I meant to make everything Lambda, I had my code prettification settings incorrectly for that file, sorry for the mistake. I'll edit and squash later today.

File: docs/_assets/js/api-search.js
Comment (RedYetiDev) [reply docs/_assets/js/api-search.js:36]: I'm a little biased (I wrote the change, so there is bias), but I find it easier to read. To me, ternary allows for more code in less space, and it keeps the statements concise and organized. If you think that an `if/then` statement is better, I'll change it back, but I prefer ternary.

**Comment by RedYetiDev** [comment]
Once the changes are approved, If you'd like, I'll squash before merging

File: docs/_assets/js/api-search.js
Comment (jchyb) [reply docs/_assets/js/api-search.js:36]: Let's at least bring back both of the comments that were in the if else before

Comment (RedYetiDev) [reply docs/_assets/js/api-search.js:36]: Right! If we are adding the comments, then you are right and the if/then may be prettier.

**Comment by RedYetiDev** [comment]
Okay, would you like me to squash before we merge?

**Comment by jchyb** [comment]
That's not necessary, I can squash while merging using GitHub UI

**Comment by RedYetiDev** [comment]
Sounds good!

**Review by jchyb — APPROVED** [review]

**Pull request merged**
